### PR TITLE
Porfolio Website: Pop-up-window

### PR DESCRIPTION
**Pull request by evansnyamekye**
###In this PR.
- I was able to create pop-up window for each project.
- I was able to remove all the hard coded code and create them dynamically from javascript file.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -60,10 +61,10 @@ <h1>I’m Nyamekye. <br>
...
   <section class="work" id="portfolio">
-    <div class="container">
-      <div class="card">
+    <div class="container wrapper">
+      <div class="card" id="1">
         <div class="card-left">
-          <img src="images/work/Snapshoot Portfolio.png" alt="snap-card">
+          <img src="images/desktop-switch/(1)Snapshoot desktop.png" alt="snap-card">
         </div>
         <div class="card-right">
```
Comment (Qoosim) [inline index.html]: - [ ] It is required that already created HTML code for the projects list should be removed, and instead it should be added via JavaScript. kIndly remove all the code for projects list here and create them dynamically on page load from your javascript file.
- [ ] Both the projects list on the page and the popup window must be created dynamically via javascript.


### Project Requirement
project-rqt

File: main.js [hunk 2]
```diff
@@ -24,3 +26,113 @@ window.onresize = () => {
...
+          wrapper.appendChild(popupPage);
+          const closeButton = popupPage.querySelector('.close-btn');
+          closeButton.addEventListener('click', () => {
+            wrapper.removeChild(popupPage);
+          });
+        }
+      }
+    });
+  });
+}
```
Comment (Qoosim) [inline main.js]: - [ ] It is required to stick to the Figma design as close as possible. Your design for the popup window on both screen sizes are different from how they appear on the figma file. Kindly give a closer look into the figma designs and walk your designs alongside the figma design.

### Your Design Desktop

your-popup-desktop


### Figma Design Desktop

temp-popup-desktop


### Your Design Mobile

your-popuu-mobile


### Figma Design Mobile

temp-popup-mobile

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi @team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct GitHub Flow. :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: index.html
Comment (aamirkhan2478) [inline index.html:1]: https://github.com/evansnyamekye/Portfolio-CompleteWebsite/blob/df73ca0a5ba30c23a7512593153a200ec183f845/index.html#L63-L161

- [ ] As mentioned by the previous reviewer :point_up_2: please add these cards dynamically inside the `JS` file but still your cards are not dynamically so kindly fix this issue :slightly_smiling_face:

File: main.js [hunk 3]
```diff
@@ -93,39 +96,39 @@ if (portfolioItems) {
...
+                <p>${portfolioItemsDetail[i].text}</p>
+             
+                <ul class="lang">
+                ${portfolioItemsDetail[i].workStack.map((stack) => `<li>${stack}</li>`).join('')}
+                </ul>
+                  <div class="link-button">
+                    <button class="btn-default"><a href="${portfolioItemsDetail[i].live} class=" see-btn">See live<img
+                    src="images/see-images/live.png" alt="live-link" /></a></button>
+                    <button class="btn-default"><a href="${portfolioItemsDetail[i].source} class=" see-btn">See source<img
+                    src="images/see-images/Vector.png" alt="live-link" /></a></button>
```
Comment (aamirkhan2478) [inline main.js:157]: - [ ] As mentioned by the previous reviewer :point_up_2: please stick to the Figma design as much as possible but your design is still not the same as the Figma design :+1:  check below :point_down:  my video for more details :slightly_smiling_face: 

[screen-capture (8).webm](https://github.com/evansnyamekye/Portfolio-CompleteWebsite/assets/45226848/fe321270-f0f8-4b45-9f73-4c12affcec7d)

| Desktop Figma Design| Mobile Figma Design |
|--------|--------|
| Desktop | Mobile |

**Review by aamirkhan2478 — CHANGES_REQUESTED** [review]
Hi @evansnyamekye & @aamir-asaram , 



Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

 ## Highlights
✔️ Good Effort
✔️ Your design is good.
✔️ All linter passed

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: index.html
Comment (evansnyamekye) [reply index.html:1]: @ aamirkhan2478 Thank you  for the review. This PR was dynamically done in main.js. Am not too sure of the branch you checked. Kindly check pop-up-menu branch that contains every js file.

Comment (evansnyamekye) [reply index.html:1]: @aamirkhan2478 Thank you for the review. This PR was dynamically done in main.js. Am not too sure of the branch you checked. Kindly check pop-up-menu branch that contains every js file.

File: main.js (hunk 3 above)
Comment (evansnyamekye) [reply main.js:157]: https://github.com/evansnyamekye/Portfolio-CompleteWebsite/assets/60971417/ae6b75eb-0110-412c-b943-44a846d1f0d2

Comment (evansnyamekye) [reply main.js:157]: @aamirkhan2478

**Review by nicupop729 — COMMENTED** [review]
Hello @evansnyamekye  👋,

_This is an additional review as a result of the second opinion request._

Great job on the remarkable effort put into this milestone.

Upon reviewing your project, I observed that the pop-up is correctly displayed on the UI, although it is currently positioned at the bottom of the screen. However, I am confident that you will address this aspect in due course, particularly if you plan to utilize this portfolio for job applications. The subsequent reviewer will not assess this specific aspect, considering it as correctly implemented.

Unfortunately, the other aspect highlighted by both code reviewers is of great significance and has not yet been implemented.

The work cards should be entirely removed from the index.html file and generated dynamically using JavaScript, just like you did with the pop-up.
To be precise, lines 65 to 161 in the index.html file should be deleted and replaced with dynamic generation.
Refer to the project requirements screenshot below:

image

Once this project requirement has been implemented, please utilize the dashboard to request another review.

Keep rocking 🚀!

**Review by Munsa1 — APPROVED** [review]
Hi @evansnyamekye, 
congratulations-2

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Made Requested Changes :heavy_check_mark: 
Correct GitHub Flow :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

## Optional suggestions
- [ ] Kindly consider to remove the commented line of code so as to achieve a clean codebase
image

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by brainconnect93 — COMMENTED** [review]
Hi @evansnyamekye,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.



## Invalid Code Review Request

This Project milestone is a pair-programming activity and it would be okay for you to create a new branch or repo and work with your new partner, only then your project can be approved. Thank you

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

_Cheers and Happy coding!👏👏👏_

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

-----

_Invalid Code Review Request does not count into the code reviews limit._

**Pull request merged**